controllers/cluster_controller.go

```diff
@@ -386,6 +388,10 @@ func (r *ClusterReconciler) preClusterProviderReconcile(ctx context.Context, log
 		}
 	}
 
+	if err := r.packagesClient.UpdateSecrets(ctx, r.client, cluster); err != nil {
```

Comment:
could this be part of the `packagesReconcile` method?

Reply:
So one of the issues I found was that the controller's `reconcile` method sometimes returns before it reaches `packagesReconcile` which is why I put it here to ensure the secret gets updated.

Reply:
I see. And what would be the problem on waiting until reconcile doesn't return early?